public/app/app.ts

```diff
@@ -205,7 +205,8 @@ export class GrafanaApp {
       setDataSourceSrv(dataSourceSrv);
       initWindowRuntime();
 
-      if (contextSrv.user.orgRole !== '') {
+      // Do not pre-load apps if improveRendererTraffic is true and the request comes from the image renderer
+      if (contextSrv.user.orgRole !== '' && !(config.featureToggles.improveRendererTraffic && contextSrv.user.authenticatedBy === 'render')) {
```

Comment:
nit: I would create a const with a meaningful name and move away these booleans from this condition.
Something like `skipAppPluginsPreload`.